src/main/java/mate/dao/jdbc/ManufacturerJdbcImpl.java

```diff
@@ -0,0 +1,118 @@
+package mate.dao.jdbc;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.dao.ManufacturerDao;
+import mate.exception.DataProcessingException;
+import mate.lib.Dao;
+import mate.model.Manufacturer;
+import mate.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerJdbcImpl implements ManufacturerDao {
+
+    private static Optional<Manufacturer> createNewManufacturer(ResultSet resultSet) {
+        try {
+            Manufacturer manufacturer = new Manufacturer(resultSet.getString("manufacturer_name"),
+                    resultSet.getString("manufacturer_country"));
+            manufacturer.setId(resultSet.getLong(1));
+            return Optional.of(manufacturer);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create new manufacturer", e);
+        }
+    }
+
+    @Override
+    public Manufacturer add(Manufacturer manufacturer) {
+        String query = "INSERT INTO manufacturers(manufacturer_name,"
+                + " manufacturer_country) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, manufacturer.getName());
+            statement.setString(2, manufacturer.getCountry());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                manufacturer.setId(resultSet.getLong(1));
+            }
+            resultSet.close();
```

Comment:
it is optional to close ResultSet, because when a Statement object is closed, its current ResultSet object, if one exists, is also closed.
https://docs.oracle.com/en/java/javase/11/docs/api/java.sql/java/sql/Statement.html#close()

Reply:
good, thank you)